web/src/inbox_ui.ts

```diff
@@ -290,6 +291,7 @@ function format_dm(
         is_bot = people.get_by_user_id(recipient_ids[0]).is_bot;
         user_circle_class = is_bot ? false : buddy_data.get_user_circle_class(recipient_ids[0]);
     }
+    const has_unread_mention = Boolean(unread.get_counts().mention_dict.get(user_ids_string));
```

Comment:
This introduces a serious performance problem, where you're fetching the entire unread history for the organization when rendering each individual row. We already called `get_counts` once as part of preparing this view and thus were able to pass in `unread_count`; you just need to compute this in the caller and pass it in.